### PR TITLE
[login] module upgraded for react

### PR DESCRIPTION
**Pull request by maltheism**
### Brief summary of changes

Upgraded module to use react. I had to fix some things in LORIS to get this all to work and there are multiple changes in some JSX files as well.

What was updated to react:
Login, password expired, request account, and reset password.

Summary:
Turns out the Content-Security policy was broken for google captcha and I made a fix to include the iframe that recaptcha uses. Major was also broken inside the php/libraries/Email.class.inc and I made some fixes. That had to do with the password reset.

Note:
This PR includes the bugfix PR #4386 and so this PR can function.

### This resolves issue...

- [ ] Redmine? [#16050](https://redmine.cbrain.mcgill.ca/issues/16050)

### To test this change...

- [x] Please test on your VM.

### CODE REVIEW

**Comment by maltheism** [comment]
Hi @kongtiaowang I made a new PR instead of rebasing the existing one. Please help with travis when you have time.

File: modules/login/php/ajaxlogin.class.inc [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+
+        try {
+            // Check if valid email.
+            if ($user->isEmailValid()) {
+                $config = \NDB_Config::singleton();
+                // Generate a new password.
+                $password = \Utility::randomString();
+                // Reset the password in the database and
+                // expire password so user must change it upon login.
+                \User::updatePassword(
```
Comment (johnsaigle) [inline modules/login/php/ajaxlogin.class.inc]:
```suggestion
                $user->updatePassword(
```

File: modules/login/jsx/passwordExpiry.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+          const state = Object.assign({}, this.state);
+          state.form.error.toggle = true;
+          state.form.error.message = data.error;
+          this.setState(state);
+        } else {
+          // success - refresh page and user is logged in.
+          window.location.href = window.location.origin;
+        }
+      }).catch((error) => {
+      // no error should happen.
```
Comment (johnsaigle) [inline modules/login/jsx/passwordExpiry.js]: But what if it does?

File: modules/login/jsx/passwordExpiry.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        class={'col-xs-12 col-sm-12 col-md-12 text-danger'}
+      />
+    ) : null;
+    const success = !this.state.request ? (
+      <div>
+        <div>
+          <p><b>Password Strength Rules</b></p>
+          <ul>
+            <li>The password must be at least 8 characters long</li>
+            <li>The password must contain at least 1 letter, 1 number and 1 character from !@#$%^*()</li>
```
Comment (johnsaigle) [inline modules/login/jsx/passwordExpiry.js]: This is out of date. Can you change it to the description I added in #4414?

File: modules/login/php/ajaxlogin.class.inc [hunk 4]
```diff
@@ -0,0 +1,386 @@
+<?php
```
Comment (johnsaigle) [inline modules/login/php/ajaxlogin.class.inc]:
```suggestion
<?php declare(strict_types=1);
```

File: modules/login/php/ajaxlogin.class.inc [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+    /**
+     * Processes the request values.
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     */
+    function processRequest(array $values) : array
+    {
+        $response = array();
```
Comment (johnsaigle) [inline modules/login/php/ajaxlogin.class.inc]: I think you should add the following here:

```php
if (!isset($values['command'])) {
   // do some sort of error handling + early return here
}
```

And below this you can remove all the `isset` checks and proceed knowing that it's good.

**Comment by maltheism** [comment]
Closing PR in favour of #5828 rebase wasn't working.

**Pull request closed without merging**